### DISCUSSION TITLE
Is there any known APIs or ways to identify between code-server (running in browser) and VSCode (running on desktop)?

### DISCUSSION THREAD

**Discussion by rfc1941** [coder/code-server#5082] [Q&A]
Dear developers,
Hello!
I was developing a extension for both code-server and desktop vscode. Due to the requirement of the extension, we need to identify if the runtime of the extension is code-server (browser) or VSCode (desktop application).  
I've gone through google and vscode API documentations and no luck. So I came to ask about this.
Thanks everyone for helping!

Effort so far:
`vscode.version` (does not work as the version of code-server is identical to that on desktop)

King Regard

**Comment by lg-nineoneninethreenine** [comment]
I don't think there is an API for code-server. However, you could just use if else conditions and send pings to the system so that it finds the session(code-server/VSCode)

**Comment by jsjoeio** [comment]
I feel like you should be able to achieve this through the [VS Code API for Web Extensions](https://code.visualstudio.com/api/extension-guides/web-extensions) maybe? I guess you need some check like:
```typescript
if (RUNTIME === "web") {
} else {
}
```

Argh...I'm not sure to be honest 🤔 Here is [how they do it upstream](https://github.com/microsoft/vscode/blob/main/src/vs/base/common/platform.ts#L78-L87).
